ExampleApp/ExampleApp/Common/Services/RxBluetoothKitService.swift

```diff
@@ -117,19 +117,28 @@ final class RxBluetoothKitService {
     // When you discover a service, first you need to establish a connection with a peripheral. Then you call
     // discoverServices(:_) for that peripheral object.
     func discoverServices(for peripheral: Peripheral) {
-        let disposable = peripheral.establishConnection()
-        .do(onNext: { [weak self] _ in
-            self?.observeDisconnect(for: peripheral)
-        })
-        .flatMap {
-            $0.discoverServices(nil)
-        }.subscribe(onNext: { [weak self] services in
+        let isConnected = peripheral.isConnected
+        
+        let createConnectedObservable = { return peripheral.discoverServices(nil).asObservable() }
+        let createDisconnectedObservable = {
```

Comment:
I'm not sure about the naming here. Firstly, you shouldn't use verbs in the names of variables, secondly both 'createConnectedObservable' and 'createDisconnectedObservable' are confusing and don't tell a lot about what's really going on. Maybe try something like 'connectedPeripheralObservable' and 'newPeripheralConnectionObservable' ?

Reply:
hmm yeah. it is a block so I would name it something like `connectedObservableCreator` and `connectObservableCreator`